doc/api/permissions.md

```diff
@@ -0,0 +1,457 @@
+# Permissions
+
+Permissions can be used to control what system resources the
+Node.js process has access to or what actions the process can take
+with those resources. Permissions can also control what modules can
+be accessed by other modules.
+
+* [Process-based permissions](#process-based-permissions) control the Node.js
+  process's access to resources such as the file system or the network.
+  The resource can be entirely allowed or denied, or actions related to it can
+  be controlled; for example, you can allow file system reads while denying
+  writes.
```

Comment:
This text should be part of the PR that includes this functionality; we can’t land the docs for that PR before the feature itself lands. Do you want to set this PR as draft maybe, and make a new PR based on this one that just has the docs renaming/reorganizing but without the “process-based permissions” additions?

Reply:
See https://github.com/nodejs/node/pull/44222#discussion_r952785571. I've included it just to reference how it gonna look like when the Permission System arises. Surely I'll remove it once we agree on the document structure -- in other words, all of the content related to process-based permissions will be removed before landing this PR